pkg/crc/preflight/preflight_checks_network_linux.go

```diff
@@ -275,6 +276,20 @@ func checkNetworkManagerIsRunning() error {
 }
 
 func checkSystemdResolvedIsRunning() error {
+	resolvFilePath := "/etc/resolv.conf"
+	// Check /ETC/RESOLV.CONF section in `man systemd-resolved`
+	systemdResolvedManageResolveFilePath := []string{"/run/systemd/resolve/stub-resolv.conf",
+		"/usr/lib/systemd/resolv.conf",
+		"/run/systemd/resolve/resolv.conf"}
+	// Check if /etc/resolv.conf is managed by systemd-resolved
+	rFilePath, err := filepath.EvalSymlinks(resolvFilePath)
+	logging.Debugf("resolve conf file path: %s", rFilePath)
+	if err != nil {
+		return err
+	}
+	if !slices.Contains(systemdResolvedManageResolveFilePath, rFilePath) {
+		return fmt.Errorf("%s not managed by systemd-resolved", resolvFilePath)
+	}
 	return checkSystemdServiceRunning("systemd-resolved.service")
```

Comment:
I'd first check if `systemd-resolved.service` is running, and if it is, then check if it manages /etc/resolv.conf.

Reply:
@cfergeau whouldn't be good to check first if `/etc/resolv.conf` managed by systemd-resolved because if it doesn't then we don't need to check the `systemd-resolved` service also so one less call?

Reply:
The argument works both ways, if systemd-resolved is not running, then there's no point in checking for the symlink, so one less call. The way I read the current code is that we first check if we are using systemd-resolved in "stub" or "foreign" mode, and after we have made this check, we verify if checking resolv.conf made any sense/was useful at all (ie are we running systemd-resolved?)

Reply:
> The argument works both ways, if systemd-resolved is not running, then there's no point in checking for the symlink, so one less call. 

no, in CSB usecase service was running but no symlink so in this case we don't have to make second call also do you think if resolved service is not running then there is any way the resolv.conf can be symlink to any of those files?

Reply:
> if the systemd-resolved service is not running, then there won't be a symlink. In this scenario your version of the code 1) checks if there's a symlink, and only then 2) checks if systemd-resolved is running.

No, in this version it will only check the symlink and since there won't be any, return the error and  it will not check systemd-resolved service so only one check.

Reply:
Ah right. The number of checks in preflights are not important anyway, code that makes sense/is readable is more important. *If* we want to improve preflight performance, this won't be done by avoiding one call in one preflight